charmstore/fakeclient.go

```diff
@@ -0,0 +1,582 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmstore
+
+import (
+	"bytes"
+	"crypto/sha512"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/api/charms"
+	"github.com/juju/juju/core/model"
+	"gopkg.in/juju/charm.v6"
+	charmresource "gopkg.in/juju/charm.v6/resource"
```

Comment:
Why `charmresource`?

Reply:
Was following some precedent from elsewhere in the codebase. Have reverted.